rfcs/0010-custom-system-uris.md

```diff
@@ -0,0 +1,325 @@
+If an application wanted to select the `Patient.identifier` entry that had a beneficiary's MBI,
+  they would search for the entry with a `system` value of `http://hl7.org/fhir/sid/us-mbi`.
+HL7, a standards body, has said that `system` should be used for all MBIs,
+  in all FHIR-compliant APIs, and so BFD does not need a custom URI there.
+If, however, an application wants to select the `Patient.identifier` entry that contains a beneficiary's CCW beneficiary ID,
+  they would search for the entry with a `system` value of `https://bluebutton.cms.gov/resources/variables/bene_id`.
+The "CCW beneficiary ID" field is only used within/by CMS,
+  so no standards body has assigned it a standard URI,
+  and BFD must provide a custom URI for it.
+
+
+## Table of Contents
+[Table of Contents]: #table-of-contents
+
+* [RFC Proposal](#rfc-proposal)
+* [Table of Contents](#table-of-contents)
+* [Motivation](#motivation)
+* [Proposed Solution](#proposed-solution)
+    * [Proposed Solution: Detailed Design](#proposed-solution-detailed-design)
+    * [Proposed Solution: Unresolved Questions](#proposed-solution-unresolved-questions)
+    * [Proposed Solution: Drawbacks](#proposed-solution-drawbacks)
+    * [Proposed Solution: Notable Alternatives](#proposed-solution-notable-alternatives)
+* [Prior Art](#prior-art)
+* [Future Possibilities](#future-possibilities)
+* [Addendums](#addendums)
+
+
+## Motivation
+[Motivation]: #motivation
+
+<!--
+Why are we doing this?
+What use cases does it support?
+What is the expected outcome?
+Why is now the appropriate time to address this?
+-->
+
+At the end of the day here,
+  our goal is to arrive at an API design decision that makes things easiest for our users.
+We may have other, internal, concerns (mostly organizational/political),
+  but providing the best possible API to our users is our primary concern.
+
+Along those lines, it's very important to note that...
+  this is not an important issue to our users,
+  unless we do something that makes their life more difficult.
+
+From the perspective of their applications,
+  our URIs are just magic strings that they will copy-paste into their code.
+From the perspective of their engineers when building out their application,
+  it's helpful for our URIs to be descriptive,
+  such that they can guess at what field they correspond to.
+From the perspective of their business decision makers,
+  the only thing that really matters is that these URIs are stable:
+  that we do not change our API in ways that are backwards-incompatible with their existing applications.
+That's about the sum total of our users' concerns:
+
+* Ensure that our URIs are documented,
+    such that they can be copy-pasted into their application code.
+* Ensure that our URIs are descriptive,
+    such that engineers building applications against our APIs can make reasonable guesses as to their meaning.
+* Ensure that our APIs continue to work with their applications.
```

Comment:
Again, there is also the issue if their validators are unable to validate the code within the specified 'system' url.

Reply:
Per Brady's statement, which implies validators unable to validate, the reason for this is that we really don't have publicly (http) accessible services that resolve a given URI.

Reply:
Resolving this also, per previous thread.